### ISSUE TITLE
deprecate repmat() ?

### ISSUE THREAD

**Issue by bjarthur** [JuliaLang/julia#20495]
given `repeat`, should `repmat` be deprecated? the latter cannot handle more than 2 dimensions, and does not have the flexibility provided by the former's inner argument.  happy to draft a PR if there is consensus.

**Comment by fredrikekre** [comment]
`repeat` is ~1000 times slower than `repmat` and uses ~40 times more memory:

```julia
julia> using BenchmarkTools

julia> for N in (10, 50, 100)
           A = rand(N, N)
           outer = (N, N)
           @btime repmat($A, $N, $N)
           @btime repeat($A, outer = $outer)
       end
  11.313 μs (202 allocations: 84.45 KiB)
  10.518 ms (119017 allocations: 3.11 MiB)
  5.407 ms (5002 allocations: 47.84 MiB)
  7.397 s (79971519 allocations: 1.98 GiB)
  106.325 ms (20002 allocations: 763.55 MiB)
  112.596 s (1294889019 allocations: 31.96 GiB)
```

**Comment by JeffBezanson** [comment]
ref #15553

**Comment by oscardssmith** [comment]
As a temporary solution, could repmat be deprecated, and a fast path added that uses the current repmat code when applicable?

**Comment by nalimilan** [comment]
`repmat` is faster, but its return type is also inferrable, which isn't currently the case for `repeat`. So I'm afraid we'll have to keep it until the compiler improves. Cf. https://github.com/JuliaLang/julia/pull/14082

**Comment by johnmyleswhite** [comment]
Would the return type be inferrable if we only allowed tuples as arguments for `inner` and `outer` and didn't use keyword arguments?

**Comment by timholy** [comment]
Yes, with those constraints one could make it inferrable.

**Comment by nalimilan** [comment]
This discussion already happened at https://github.com/JuliaLang/julia/issues/12953 and https://github.com/JuliaLang/julia/pull/14082, and it was decided to keep keyword arguments.

**Comment by JeffBezanson** [comment]
Closed by #26039
